liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -1883,28 +1884,11 @@ public final void dropAll(CatalogAndSchema... schemas) throws DatabaseException
      * 'Tags' the database for future rollback
      */
     public void tag(String tagString) throws LiquibaseException {
-        runInScope(new Scope.ScopedRunner() {
-            @Override
-            public void run() throws Exception {
-
-                LockService lockService = LockServiceFactory.getInstance().getLockService(database);
-                lockService.waitForLock();
-
-                try {
-                    ChangeLogHistoryServiceFactory.getInstance().getChangeLogService(database).generateDeploymentId();
-
-                    checkLiquibaseTables(false, null, new Contexts(),
-                            new LabelExpression());
-                    getDatabase().tag(tagString);
-                } finally {
-                    try {
-                        lockService.releaseLock();
-                    } catch (LockException e) {
-                        LOG.severe(MSG_COULD_NOT_RELEASE_LOCK, e);
-                    }
-                }
-            }
-        });
+        CommandScope tagCommand = new CommandScope("tag");
+        tagCommand
+                .addArgumentValue(TagCommandStep.DATABASE_ARG, database)
+                .addArgumentValue(TagCommandStep.TAG_ARG, tagString);
+        tagCommand.execute();
```

Comment:
Are we keeping this method for backwards compatibility?

Reply:
Yes, people using Liquibase as a library rely on that. But thinking about it now, maybe it should be deprecated?

Reply:
Marked as deprecated